src/main/java/com/minecolonies/coremod/colony/buildings/modules/WorkAtHomeBuildingModule.java

```diff
@@ -46,8 +46,8 @@ public boolean assignCitizen(final ICitizenData citizen)
                         buildingComponent,
                         BlockPosUtil.getString(oldHome.getID()))
                       .sendTo(oldHome.getColony()).forAllPlayers();
+                    oldHome.getFirstModuleOccurance(LivingBuildingModule.class).removeCitizen(citizen);
                 }
-                oldHome.getFirstModuleOccurance(LivingBuildingModule.class).removeCitizen(citizen);
```

Comment:
This seems suspicious.  Won't this leave the citizen assigned to two modules if the `if` condition fails (e.g. the old home does have a `WorkAtHomeBuildingModule`)?

Reply:
if (oldHome.hasModule(LivingBuildingModule.class) && !oldHome.hasModule(WorkAtHomeBuildingModule.class))

I think the second part of this if condition shouldn't be. Else assigning from one guard tower to another can cause issues.

Reply:
That part of the `if` condition looks to be fine for the chat message (saying that a house has a free bed now), possible not for the rest of the block statement

Reply:
yeah, should have the oldHome has module for the remove check, and the second one for the print message